File: src/main/java/com/cfgglobal/test/domain/Rule.java
package com.cfgglobal.test.domain;


import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Type;

import javax.persistence.Entity;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@ToString
@Accessors(chain = true)
@DynamicUpdate
@DynamicInsert
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Rule extends BaseEntity {


    String name;

    String params;

    String type;

    @Type(type = "yes_no")
    @NotNull
    Boolean enable = true;

}

File: src/main/kotlin/com/github/leon/aci/domain/Rule.kt
package com.github.leon.aci.domain


import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import org.hibernate.annotations.Type
import javax.persistence.Entity
import javax.persistence.Table
import javax.validation.constraints.NotNull

@Entity
@DynamicUpdate
@DynamicInsert
@Table(name = "aci_rule")
data class Rule(

        @NotNull
        val name: String = "",

        @NotNull
        val params: String? = null,

        @NotNull
        val type: String? = null,

        @Type(type = "yes_no")
        @NotNull
        val enable: Boolean = true

) : BaseEntity()
